modules/runner/src/main/java/org/apache/ignite/internal/app/IgnitionImpl.java

```diff
@@ -233,8 +242,8 @@ else if (cfgContent != null)
     /**
      * Starts the Vault component.
      */
-    private static VaultManager createVault(String nodeName) {
-        Path vaultPath = VAULT_DB_PATH.resolve(nodeName);
+    private static VaultManager createVault(String nodeName, Path workDir) {
```

Comment:
I think that the `workDir`, which we receive from the outside, should be already the path to the directory for this concrete node, and so workDir.resolve(VAULT_DB_PATH) will be enough.